xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/gallery/gallery.js

```diff
@@ -22,25 +22,29 @@ var XWiki = (function (XWiki) {
 XWiki.Gallery = Class.create({
   initialize : function(container) {
     this.images = this._collectImages(container);
-
     this.container = container.update(
-      '<input type="text" tabindex="-1" class="focusCatcher"/>' +
+      '<button class="maximize" title="${escapetool.xml($services.localization.render("core.widgets.gallery.maximize"))}"></button>' +
+      '<button class="previous" title="${escapetool.xml($services.localization.render("core.widgets.gallery.previousImage"))}">&lt;</button>' +
       '<img class="currentImage" alt="${escapetool.xml($services.localization.render("core.widgets.gallery.currentImage"))}"/>' +
-      '<div class="previous" title="${escapetool.xml($services.localization.render("core.widgets.gallery.previousImage"))}">&lt;</div>' +
-      '<div class="next" title="${escapetool.xml($services.localization.render("core.widgets.gallery.nextImage"))}">&gt;</div>' +
-      '<div class="index">0 / 0</div>' +
-      '<div class="maximize" title="${escapetool.xml($services.localization.render("core.widgets.gallery.maximize"))}"></div>'
+      '<button class="next" title="${escapetool.xml($services.localization.render("core.widgets.gallery.nextImage"))}">&gt;</button>' +
+      '<div class="index" tabindex="0" title="${escapetool.xml($services.localization.render("core.widgets.gallery.index.description"))}" aria-description="${escapetool.xml($services.localization.render("core.widgets.gallery.index.description"))}">0 / 0</div>'
     );
-    this.container.addClassName('xGallery');
-
-    this.focusCatcher = this.container.down('.focusCatcher');
+    this.container.addClassName('xGallery');    
+    
+    /* Instead of an arbitrary element to catch focus, we use the index.
+    * This index already stores the current image state, might as well be responsible for providing quick controls and
+    * explanations about these quick controls.
+    * Note that wrapping the image in an interactive container to handle this would have been a good solution too.
+    * However, this wrapping caused the image to overflow the CSS grid vertically when in maximized mode. 
+    * Technically I couldn't find a CSS solution to prevent this, so I decided to make do without wrapping. */
```

Comment:
```suggestion
    // Instead of an arbitrary element to catch focus, we use the index.
    // This index already stores the current image state, might as well be responsible for providing quick controls and
    // explanations about these quick controls.
    // Note that wrapping the image in an interactive container to handle this would have been a good solution too.
    // However, this wrapping caused the image to overflow the CSS grid vertically when in maximized mode. 
    // Technically I couldn't find a CSS solution to prevent this, so I decided to make do without wrapping.
```

Multiline comments are normally used to document functions (methods) and classes (or types in general).

Reply:
We don't have a code style for javascript AFAIR. I assumed it'd be best to use the [same comment style as the one we use for CSS](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/). It would be interesting to keep trace of this rule so that new contributors won't make the same mistake. I created the draft doc https://dev.xwiki.org/xwiki/bin/view/Drafts/Javascript%20Code%20Style/ for now.

[Reference of where we can already see this codestyle in action](https://github.com/xwiki/xwiki-platform/blob/46ef42cc41958a24c5c58fca20942c0beeec6b69/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/viewers/attachments.js#L54-L60)

Reply:
> I assumed it'd be best to use the [same comment style as the one we use for CSS](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/).

JavaScript is a programming language, unlike CSS, so I don't think it makes sense to look at CSS code styles for defaults. The default should be the **Java** code style IMO.